### ISSUE TITLE
Development branch confusion

### ISSUE THREAD

**Issue by gravitystorm** [cyclestreets/cyclescape#592]
Ok, I'm confused with the development branches!

The way it used to work was that we created feature branches, then when they were ready, merged into staging. Staging was then merged into master when we deployed. The only reason we didn't merge feature branches straight into master was to avoid deploying multiple times per day.

From the last few months of commits, there are many which appear to be made directly to master, and only at some point later merged into staging. Meanwhile, big changes are being made to staging. Staging hasn't been merged into master since April so there's a ton of stuff that hasn't been deployed, and other things have been deployed since (by committing directly to master).

What's going on? Which branch should I make pull requests against? When is staging next going to be merged into master?

**Comment by nikolai-b** [comment]
Hi,
My fault.  I wanted to be able to show @mvl22 the leaflet work that I've done so the flow become messed up.  We thought things were nearly ready so merged into staging but after more testing found issues.  When everyone is happy with the ripping out of Openlayers and replacing it the leaflet then staging can be merged into master and the confusion will cease.  Till then urgent issues are committed directly to master and general development can go onto staging.
Sorry again about the mess!

**Comment by gravitystorm** [comment]
Fair enough! Thanks for the details.

I'd like to get some things from staging deployed ASAP so I'm keen to help. Do you have a list of things that need doing before merge? I also noticed a few bugs when using staging locally, so I'll either log those or make PRs for them too.

Is the plan to complete the leaflet migration entirely before deployment, or are you happy to deploy with a mixture of OpenLayers and Leaflet?

**Comment by nikolai-b** [comment]
I've worked through the things I found and I'm generally happy, I guess the final thing is to delete the OpenLayers js code and have another look at the site.  Ideally someone with fresh eyes would give it a once over and see if we are missing anything obvious so if you wanted to that'd be great.
Martin is away this weekend but when he's back I was hoping to get staging into master soon.

**Comment by gravitystorm** [comment]
Great - I'll do what I can to fix any bugs that I find on staging - there's a bunch of things on there already which I really want to see deployed ASAP, especially to build momentum with our translators who aren't getting to see their work deployed.

Unfortunately I spotted a bunch of niggles with the Leaflet work (see #600 ). Unless we fix all those we might need to deploy with some niggles remaining since a lot of stuff is waiting. What does everyone think? Should we just deploy now and fix things later?

**Comment by jf1** [comment]
Looks like at least a couple of the items in #600 should be fixed first, eg
"no way to reposition a point when editing" & "drawing a route actually
draws an area".
I feel other niggles such as styling could be addressed once live,
especially if that allows progress to be made on other elements that are
already in staging.

**Comment by gravitystorm** [comment]
@jf1 I think after #606 is reviewed we'll be about ready for release. I can still see other bugs but I'd really, really like to see staging deployed ASAP.

**Comment by jf1** [comment]
Obviously this has my support, for what that's worth. Onwards!

**Comment by nikolai-b** [comment]
I agree but I'd really like @mvl22 to confirm he's happy first...

**Comment by mvl22** [comment]
Collisions checkbox box has disappeared on the issue page.

Using the hard-to-find tickbox within the standard leaflet control also doesn't show collisions, but generates error `Uncaught TypeError: Cannot read property 'lat' of undefined`.

**Comment by mvl22** [comment]
On `/user/locations/new`, if you draw a polygon, then press Clear to remove it, the `Drag handles, or marker to edit feature` tooltip is shown, but no type is selected, so it doesn't actually do anything and doesn't give the user any indication why not.

I think if Clear is chosen, the previously-selected object type should be selected. So if drawing an area, and Clear is clicked on, the Area button should get auto-selected.

**Comment by mvl22** [comment]
The namefinder input box on the left, under +/- box, is very small. I would prefer the styling to be the same or similar to:

https://www.cyclestreets.net/collisions/

**Comment by mvl22** [comment]
> Can we keep leaflet discussions on #600 instead of here?

Apologies - confusion from too many tabs open at the moment!

**Comment by gravitystorm** [comment]
Super - `staging` has been merged into `master` and deployed - thank you! I think we can now go back to normal, with pull requests being merged into staging and deployed within a few days.

**Comment by mvl22** [comment]
Yes, thanks all on this.

Could we take the opportunity to get the What's New page updated with all the changes.

**Comment by jf1** [comment]
Thanks all.

**Comment by nikolai-b** [comment]
Updated in https://github.com/cyclestreets/cyclescape/commit/f274ff6e4050f17a7b8b7e9764520956a9f56af3

**Comment by jf1** [comment]
Typo and suggestion:
tag could -> tag cloud at http://www.cyclescape.org/tags

Another idea:
Downloadable deadline ics -> Download deadlines as iCal .ics files
